tests/php/test-uninstall.php

```diff
@@ -0,0 +1,52 @@
+<?php
+/**
+ * Test cases for uninstall.php
+ *
+ * @package AMP
+ */
+
+/**
+ * @runInSeparateProcess
+ * @group uninstall
+ */
+class Test_Uninstall extends WP_UnitTestCase {
+
+	/**
+	 * @covers ::delete_options
+	 * @covers ::delete_posts
+	 * @covers ::delete_terms
+	 */
+	public function test_uninstall_php() {
+
+		/**
+		 * Create dummy data.
+		 */
+
+		update_option( 'amp-options', 'Yes' );
+
+		$amp_validated_post = $this->factory()->post->create_and_get(
+			[
+				'post_type' => \AMP_Validated_URL_Post_Type::POST_TYPE_SLUG,
+			]
+		);
+
+		$amp_error_term = $this->factory()->term->create_and_get(
+			[
+				'taxonomy' => 'amp_validation_error',
+			]
+		);
```

Comment:
Please also create a `post`, create a `post_tag`, update the `blog_name`, and create a transient here. Then after `uninstall.php` make sure that these are remain. 

We need to make extra super careful that we do not touch any data that isn't part of the AMP plugin.

Reply:
Thanks @westonruter 

I have updated the unit test cases, Please have a look.